src/hooks/use-slides.tsx

```diff
@@ -12,6 +12,8 @@ export function useCollectSlides() {
   const [initialized, setInitialized] = useState(false);
   const [slideContainer, setSlideContainer] = useState<HTMLElement | null>();
   const [slideIds, setSlideIds] = useState<SlideId[]>([]);
+  const [slideIdsOfSlidesWithTemplates, setSlideIdsOfSlidesWithTemplates] =
+    useState<SlideId[]>([]);
```

Comment:
It looks like the primary use-case for this state is to check whether or not a slide ID has a template, and we don't care so much about order of IDs etc.. In this case, I recommend we use a `Set` instead of an array, so that our "contains" lookup is `O(1)` complexity instead of `O(n)`.

Reply:
Good idea, have updated to be a Set in: https://github.com/FormidableLabs/spectacle/pull/1116/commits/7a336940c61e50bda402852eaa5c790a4cdb1a38